### PR TITLE
Port crypto/ to Python 3

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

**Comment by hawkowl** [comment]
@matrixbot retest this please

**Comment by hawkowl** [comment]
@matrixbot retest this please

File: synapse/crypto/keyclient.py [hunk 1]
```diff
@@ -82,6 +82,12 @@ def connectionMade(self):
         self._peer = self.transport.getPeer()
         logger.debug("Connected to %s", self._peer)
 
+        if not isinstance(self.path, bytes):
+            self.path = self.path.encode('ascii')
```
Comment (erikjohnston) [inline synapse/crypto/keyclient.py]: Can paths and hosts not be unicode?

Comment (hawkowl) [reply synapse/crypto/keyclient.py]: nope, must be ascii-encoded (maybe punycode if it's a hostname or url-encoded if it's a path), because http

**Pull request merged**